eq-author/src/components/ContentPickerv2/index.js

```diff
@@ -0,0 +1,172 @@
+  position: absolute;
+  right: 0.5em;
+  top: 2em;
+  bottom: 0;
+  margin: auto;
+  align-items: center;
+`;
+
+const StyledModal = styled(Modal)`
+  .Modal {
+    padding: 0;
+    width: 45em;
+  }
+`;
+
+const Container = styled.div`
+  background: white;
+`;
+
+const ContentPicker = ({
+  contentType,
+  data,
+  onOpen,
+  onClose,
+  isOpen,
+  startingSelectedAnswers,
+  multiselect,
+  onSubmit,
+}) => {
+  const [selectedItems, setSelectedItems] = useState(startingSelectedAnswers);
+
+  const isSelected = item => {
+    return (
+      selectedItems.findIndex(selectedItem => selectedItem.id === item.id) !==
+      -1
+    );
+  };
+
+  const getFirstSelectedItemId = () => {
+    if (!selectedItems.length) {
+      return false;
+    }
+
+    return selectedItems[0].id;
+  };
+
+  const updateSelectedItemsSingle = item => {
+    const updatedSelectedItems = !item || isSelected(item) ? [] : [item];
+    setSelectedItems(updatedSelectedItems);
+  };
+
+  const updateSelectedItemsMultiple = item => {
+    const updatedSelectedItems = isSelected(item)
+      ? selectedItems.filter(selectedItem => selectedItem.id !== item.id)
+      : [...selectedItems, item];
+
+    setSelectedItems(updatedSelectedItems);
+  };
+
+  const handleConfirm = () => {
+    if (multiselect) {
+      onSubmit(selectedItems);
+    } else {
+      onSubmit(selectedItems[0]);
+    }
+  };
+
+  const handleSelected = multiselect
+    ? updateSelectedItemsMultiple
+    : updateSelectedItemsSingle;
+
+  const renderPicker = () => {
+    switch (contentType) {
+      case METADATA:
+        return (
+          <MetadataPicker
+            onConfirm={handleConfirm}
+            onSelected={handleSelected}
+            isSelected={isSelected}
+            data={data}
+          />
+        );
+
+      case ANSWER:
+        return (
+          <AnswerPicker
+            onConfirm={handleConfirm}
+            onSelected={handleSelected}
+            isSelected={isSelected}
+            data={data}
+            multiselect={multiselect}
+            firstSelectedItemId={getFirstSelectedItemId()}
+          />
+        );
+
+      default:
+        return <div data-test="content-picker-empty" />;
+    }
+  };
+
+  return (
+    <StyledModal
+      isOpen={isOpen}
+      onOpen={onOpen}
+      onClose={onClose}
+      hasCloseButton
+    >
+      <Container>{renderPicker()}</Container>
+      <ModalFooter>
+        <ButtonGroup horizontal align="right">
+          <Button variant="secondary" onClick={onClose}>
+            Cancel
+          </Button>
+          <Button
+            variant="primary"
+            autoFocus
+            onClick={handleConfirm}
+            disabled={!selectedItems.length}
+          >
+            Confirm
+          </Button>
+        </ButtonGroup>
+      </ModalFooter>
+    </StyledModal>
+  );
+};
+
+ContentPicker.propTypes = {
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
```

Comment:
Can we use `PropTypes.oneOfType([customPropTypes.Metadata, customPropTypes.Answer])` here

Reply:
All propTypes changes sorted